src/interfaces/types.ts

```diff
@@ -5,13 +5,13 @@ import type { Readable } from 'readable-stream';
 /**
  * Interface that defines a message handler of a specific method.
  */
-export interface MethodHandler {
+export interface MethodHandler<ResultEntryKind> {
   /**
    * Handles the given message and returns a `MessageReply` response.
    */
   handle(input: {
     tenant: string;
     message: BaseMessage;
     dataStream?: Readable
-  }): Promise<MessageReply>;
+  }): Promise<MessageReply<ResultEntryKind>>;
```

Comment:
@thehenrytsai , i def wasn't a fan of the generic approach specifically bc of  [this weirdness](https://github.com/TBD54566975/dwn-sdk-js/blob/events-get/src/interfaces/records/handlers/records-read.ts#L13).  it should be `MessageReply<never>` because `RecordsRead` will  _never_ have `entries`!  

but i can't use `never` without changing [this](https://github.com/TBD54566975/dwn-sdk-js/blob/events-get/src/dwn.ts#L26) from `unknown` to `any`. so end up being stuck between a rock and a hard place.

 using `unknown` implies that _something_ will for sure be there, but exactly _what_ is unknown. In the case of `RecordsRead` this is a straight up lie  because we _know_ `entries` will _never_ be there. 

whereas `any` means: "could be.. or maybe not 🤷🏽 ." neither of which necessarily seem helpful.

Reply:
@thehenrytsai i can't use the approach that you used and suggested i follow from #296 without making some changes. here's why:

the [`MethodHandler`](https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/interfaces/types.ts#L16) interface has a return signature of `Promise<MessageReply>`.  This works out fine in the case of [`RecordsReadReply`](https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/interfaces/records/types.ts#L120-L130) because `RecordsReadReply` is an intersection of `BaseMessageReply` _and_ an object with an _optional_ `record` object. the fact that `record` is _optional_ in addition to the fact that `RecordsReadReply` doesn't affect, use, or modify `events` _or_ `data` as defined in [`MessageReply`](https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/core/message-reply.ts#L19-L49) makes it so that the return signature of [`RecordsReadHandler.handle`](https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/interfaces/records/handlers/records-read.ts#L20) remains adherent to `MethodHandler` interface. 

in the case of `EventsGet`, `EventsGetReply` would look something like this:
```ts
export type EventsGetReply = BaseMessageReply & {
  entries?: Event[];
};
```
_unrelated note: also fine with `entries` being required, but it isn't for `RecordsQuery` replies when an error is present iirc_

changing `EventsGetHandler.handler`'s return signature to `Promise<EventsGetReply>` violates the `MethodHandler` interface specifically because the `entries` types of `MessageReply` and `EventsGetReply` are incompatible. 

unless i'm missing something, which i might be! we'll likely have to rethink the plan mentioned in your comment.

Reply:
maybe call it `events` instead of `entries`?

Reply:
bless you kind sir. did exactly that